include/multipass/virtual_machine.h

```diff
@@ -90,15 +90,23 @@ class VirtualMachine : private DisabledCopyMove
     using SnapshotVista = std::vector<std::shared_ptr<const Snapshot>>; // using vista to avoid confusion with C++ views
     virtual SnapshotVista view_snapshots() const noexcept = 0;
     virtual int get_num_snapshots() const noexcept = 0;
+
     virtual std::shared_ptr<const Snapshot> get_snapshot(const std::string& name) const = 0;
+    virtual std::shared_ptr<const Snapshot> get_snapshot(int index) const = 0;
     virtual std::shared_ptr<Snapshot> get_snapshot(const std::string& name) = 0;
+    virtual std::shared_ptr<Snapshot> get_snapshot(int index) = 0;
+
     virtual std::shared_ptr<const Snapshot> take_snapshot(const VMSpecs& specs,
                                                           const std::string& name,
                                                           const std::string& comment) = 0;
+    virtual void rename_snapshot(const std::string& old_name, const std::string& new_name) = 0;
```

Comment:
Otherwise, this feels like we're creating another situation like how we're not being able to easily rename instances in Multipass.

Reply:
That is true, but it would require changing the map from names to IDs. At least theoretically, that would make something like `multipass info <snapshot>` very slightly slower, while internal fetches by index would be faster. The real reason though was just getting this done faster. But you are right, that would be an improvement. How about I add a `TODO@snapshots` to deal with next, as time permits?

Reply:
Yeah, I'm happy with that. Just something that would be nice to do the right way.